### PR TITLE
LPD-39899 Improvements for Permission's taglib

### PR DESCRIPTION
**Pull request by ambrinchaudhary**
# What is this trying to solve 
This is part of the story https://liferay.atlassian.net/browse/LPD-35510. This pull includes: 
- https://liferay.atlassian.net/browse/LPD-39899 Remove more/hide options button 
- https://liferay.atlassian.net/browse/LPD-39898 Change label in DM from “Viewable by” to “Viewable and Downloadable by” (only applies to File Upload and Multiple Upload in DM)

Screenshot 2024-10-23 at 14 17 50
Screenshot 2024-10-23 at 14 18 19

Other places like Blogs, Web Content ... are not affected, because DM is the only one that has de Download permission. 

# How can I test it

You can go to Content & Data > Documents and Media and add a File Upload or Multiple Files Upload. 

Or run `npm run playwright -- test -g 'LPD-39898' --reporter list`

```
npm run playwright -- test -g 'LPD-39898' --reporter list --repeat-each=5

> @liferay/playwright@1.0.0 playwright
> playwright "test" "-g" "LPD-39898" "--reporter" "list" "--repeat-each=5"


Running 5 tests using 1 worker

  ✓  1 [document-library-web] › document-library-web/dmPermissions.spec.ts:22:1 › DM Permissions initial status (27.9s)
  ✓  2 [document-library-web] › document-library-web/dmPermissions.spec.ts:22:1 › DM Permissions initial status (24.9s)
  ✓  3 [document-library-web] › document-library-web/dmPermissions.spec.ts:22:1 › DM Permissions initial status (22.2s)
  ✓  4 [document-library-web] › document-library-web/dmPermissions.spec.ts:22:1 › DM Permissions initial status (35.1s)
  ✓  5 [document-library-web] › document-library-web/dmPermissions.spec.ts:22:1 › DM Permissions initial status (22.7s)

  5 passed (2.5m)
```

**Note:** Previously approved in Content Management: https://github.com/liferay-content-management/liferay-portal/pull/6028

### CODE REVIEW

**Comment by ambrinchaudhary** [comment]
ci:test:relevant

**Review by bryceosterhaus — APPROVED** [review]
Looks good to me

**Comment by ambrinchaudhary** [comment]
ci:test:relevant

**Comment by ambrinchaudhary** [comment]
ci:report:52628102

**Comment by ambrinchaudhary** [comment]
ci:test:relevant

**Comment by ambrinchaudhary** [comment]
CI shows these as unique failures: 
```
PortalLogAssertorTest-modules-integration-mysql57/2/17
com.liferay.object.rest.internal.resource.v1_0.test.ObjectEntryResourceTest
com.liferay.segments.test.SegmentsEntryRetrieverTest
com.liferay.headless.builder.resource.test.HeadlessBuilderOpenAPIResourceTest
```

But I don't see how they could be connected to these changes. Do you think is save to force forward @bryceosterhaus?

**Comment by markocikos** [comment]
ci:report:53694528

File: portal-web/docroot/html/taglib/ui/input_permissions/page.jsp [hunk 1]
```diff
@@ -113,13 +120,15 @@ String modelName = (String)request.getAttribute("liferay-ui:input-permissions:mo
 					<option <%= inputPermissionsViewRole.equals(RoleConstants.OWNER) ? "selected=\"selected\"" : "" %> value="<%= RoleConstants.OWNER %>"><liferay-ui:message key="owner" /></option>
 				</select>
 
-				<button aria-controls="<%= uniqueNamespace %>inputPermissionsTable" aria-expanded="<%= inputPermissionsShowOptions %>" class="btn btn-secondary btn-sm mt-3" id="<%= uniqueNamespace %>inputPermissionsOptionsButton" onclick="<%= uniqueNamespace %>inputPermissionsToggle();" type="button">
-					<%= inputPermissionsShowOptions ? LanguageUtil.get(request, "hide-options") : LanguageUtil.get(request, "more-options") %>
-				</button>
+				<c:if test='<%= !FeatureFlagManagerUtil.isEnabled("LPD-19787") %>'>
+					<button aria-controls="<%= uniqueNamespace %>inputPermissionsTable" aria-expanded="<%= inputPermissionsShowOptions %>" class="btn btn-secondary btn-sm mt-3" id="<%= uniqueNamespace %>inputPermissionsOptionsButton" onclick="<%= uniqueNamespace %>inputPermissionsToggle();" type="button">
+						<%= inputPermissionsShowOptions ? LanguageUtil.get(request, "hide-options") : LanguageUtil.get(request, "more-options") %>
+					</button>
```
[the remark this answers is not in the document]
Comment (ambrinchaudhary) [reply portal-web/docroot/html/taglib/ui/input_permissions/page.jsp:126]: Sure, done!

File: portal-web/docroot/html/taglib/ui/input_permissions/page.jsp [hunk 2]
```diff
@@ -66,7 +66,7 @@ String modelName = (String)request.getAttribute("liferay-ui:input-permissions:mo
 
 		boolean submitted = request.getParameter(groupPermissionsName) != null;
 
-		boolean inputPermissionsShowOptions = ParamUtil.getBoolean(request, "inputPermissionsShowOptions");
+		boolean inputPermissionsShowOptions = FeatureFlagManagerUtil.isEnabled("LPD-19787") || ParamUtil.getBoolean(request, "inputPermissionsShowOptions");
```
[the remark this answers is not in the document]
Comment (ambrinchaudhary) [reply portal-web/docroot/html/taglib/ui/input_permissions/page.jsp:69]: I'm not sure about this. This input is send to the server and I don't know the implications that may occur if the value is not present, so for now I prefer to just send a true to this value. We can analyse it carefully when we need to remove the FF.

**Comment by ambrinchaudhary** [comment]
ci:test:sf

**Review by markocikos — APPROVED** [review]
@ambrinchaudhary LGTM!

**Comment by ambrinchaudhary** [comment]
ci:forward

**Comment by ambrinchaudhary** [comment]
ci:forward:force

**Pull request closed without merging**